Spanner/src/Connection/Grpc.php

```diff
@@ -443,13 +472,15 @@ public function testDatabaseIamPermissions(array $args)
     public function createSession(array $args)
     {
         $databaseName = $this->pluck('database', $args);
+        $databaseParsed = SpannerClient::parseName($databaseName);
+        $intanceName = InstanceAdminClient::instanceName($databaseParsed['project'], $databaseParsed['instance']);
```

Comment:
I have not yet begun to change this method, because the [same call](https://github.com/q-logic/google-cloud-php/blob/master/Spanner/src/Batch/BatchClient.php#L179-L182) is also used in the `BatchClient` class. Only there I did not find the opportunity to get the name instance.

The `batchCreateSessions` method did not edit either, because it is found in the class `CahceSessionPool` where you cannot get the session name from the database object.

`createSessionAsync` is generally found only in tests.

Reply:
Yes, it looks like that it is easier to obtain `$instanceName` right here in inside the method.